File: clients/go/examples/admin-list-databases.go
package main

import (
	"context"
	"encoding/json"
	"fmt"
	"log"

	"github.com/datastax/astra-db-go/v2/astra"
	"github.com/datastax/astra-db-go/v2/astra/options"
)

func main() {
	ctx := context.Background()

	client := astra.NewClient(
		options.API().SetToken("**APPLICATION_TOKEN**"),
	)

	admin, err := client.Admin()

	if err != nil {
		log.Fatal(err)
	}

	databases, err := admin.ListDatabases(ctx)

	if err != nil {
		log.Fatal(err)
	}

	output, err := json.Marshal(databases)

	if err != nil {
		log.Fatal(err)
	}

	fmt.Println(string(output))
}

File: clients/java/examples/admin-list-databases.java
import com.datastax.astra.client.DataAPIClient;
import com.datastax.astra.client.admin.AstraDBAdmin;
import com.datastax.astra.client.databases.definition.DatabaseInfo;
import java.util.List;

public class Example {
  public static void main(String[] args) {
    DataAPIClient client = new DataAPIClient("**APPLICATION_TOKEN**");

    AstraDBAdmin admin = client.getAdmin();

    List<DatabaseInfo> databases = admin.listDatabases();

    System.out.println(databases);
  }
}
